cli/batch-score-rest.sh

```diff
@@ -0,0 +1,311 @@
+--header "Authorization: Bearer $TOKEN" \
+--data-raw "{
+    \"location\": \"$LOCATION\",
+    \"properties\": {        
+        \"model\": {
+            \"referenceType\": \"Id\",
+            \"assetId\": \"/subscriptions/$SUBSCRIPTION_ID/resourceGroups/$RESOURCE_GROUP/providers/Microsoft.MachineLearningServices/workspaces/$WORKSPACE/models/mnist/versions/1\"
+        },
+        \"codeConfiguration\": {
+            \"codeId\": \"/subscriptions/$SUBSCRIPTION_ID/resourceGroups/$RESOURCE_GROUP/providers/Microsoft.MachineLearningServices/workspaces/$WORKSPACE/codes/score-mnist/versions/1\",
+            \"scoringScript\": \"digit_identification.py\"
+        },
+        \"environmentId\": \"/subscriptions/$SUBSCRIPTION_ID/resourceGroups/$RESOURCE_GROUP/providers/Microsoft.MachineLearningServices/workspaces/$WORKSPACE/environments/mnist-env/versions/$ENV_VERSION\",
+        \"compute\": \"/subscriptions/$SUBSCRIPTION_ID/resourceGroups/$RESOURCE_GROUP/providers/Microsoft.MachineLearningServices/workspaces/$WORKSPACE/computes/cpu-cluster\",
+        \"resources\": {
+            \"instanceCount\": 1
+        },
+        \"maxConcurrencyPerInstance\": \"4\",
+        \"retrySettings\": {
+            \"maxRetries\": 3,
+            \"timeout\": \"PT30S\"
+        },
+        \"errorThreshold\": \"10\",
+        \"loggingLevel\": \"info\",
+        \"miniBatchSize\": \"5\",
+    }
+}")
+#</create_deployment>
+# TEMPORARY! AzureAsyncOperationUri is not being returned on Deployment
+sleep 60s
+#operation_id=$(echo $response | jq '.properties.properties.AzureAsyncOperationUri')
+#wait_for_completion $operation_id
+
+#<set_endpoint_defaults>
+response=$(curl --location --request PUT "https://management.azure.com/subscriptions/$SUBSCRIPTION_ID/resourceGroups/$RESOURCE_GROUP/providers/Microsoft.MachineLearningServices/workspaces/$WORKSPACE/batchEndpoints/$ENDPOINT_NAME?api-version=$API_VERSION" \
+--header "Content-Type: application/json" \
+--header "Authorization: Bearer $TOKEN" \
+--data-raw "{
+    \"properties\": {
+        \"authMode\": \"aadToken\",
+        \"defaults\": {
+            \"deploymentName\": \"$DEPLOYMENT_NAME\"
+        }
+    },
+    \"location\": \"$LOCATION\"
+}")
+
+operation_id=$(echo $response | jq -r '.properties' | jq -r '.properties' | jq -r '.AzureAsyncOperationUri')
+wait_for_completion $operation_id
+
+#</set_endpoint_defaults>
+
+# <get_endpoint>
+response=$(curl --location --request GET "https://management.azure.com/subscriptions/$SUBSCRIPTION_ID/resourceGroups/$RESOURCE_GROUP/providers/Microsoft.MachineLearningServices/workspaces/$WORKSPACE/batchEndpoints/$ENDPOINT_NAME?api-version=$API_VERSION" \
+--header "Content-Type: application/json" \
+--header "Authorization: Bearer $TOKEN")
+
+SCORING_URI=$(echo $response | jq -r '.properties.scoringUri')
+# </get_endpoint>
+
+# <get_access_token>
+SCORING_TOKEN=$(az account get-access-token --resource https://ml.azure.com --query accessToken -o tsv)
```

Comment:
dup? TOKEN at the beginning already.

Reply:
`--resource https://ml.azure.com` does not provide access to the ARM resources, only to the ScoringUri.

A fully scoped token is required for all of the ARM calls.